pkg/cloudproviders/k8s/k8s.go

```diff
@@ -0,0 +1,380 @@
+	for _, lookupIP := range IP {
+		if lookupIP == gostatsd.UnknownIP {
+			continue
+		}
+
+		p.logger.WithField("ip", lookupIP).Debug("Looking up pod ip")
+		objs, err := p.PodsInf.GetIndexer().ByIndex(PodsByIPIndexName, string(lookupIP))
+		if err != nil {
+			instanceIPs[lookupIP] = nil
+			returnErr = err
+			continue
+		}
+		if len(objs) < 1 {
+			p.logger.Debugf("Could not find IP in cache, continuing")
+			instanceIPs[lookupIP] = nil
+			continue
+		}
+		if len(objs) > 1 {
+			p.logger.Warn("More than one pod in cache for IP '%s'. Using first stored", lookupIP)
+		}
+		pod := objs[0].(*core_v1.Pod)
+
+		// Turn the pod metadata into tags
+		tags := gostatsd.Tags{}
+		// TODO: deduplicate labels and annotations in their tag format, rather than overwriting
+		for k, v := range pod.ObjectMeta.Labels {
+			if tagKeyMatchesWhitelist(k, p.labelWhitelist) {
+				tags = append(tags, fmt.Sprintf("%s:%s", k, v))
+			}
+		}
+		for k, v := range pod.ObjectMeta.Annotations {
+			if !tagKeyMatchesWhitelist(k, p.annotationWhitelist) {
+				continue
+			}
+			// Annotations are often of the format "purpose.company.com/annotationName" so we want to expectedKey off the
+			// start when processing tag names
+			tagName := k
+			splitTagName := strings.SplitN(k, "/", 2)
+			if len(splitTagName) > 1 {
+				tagName = splitTagName[1]
+			}
+			tags = append(tags, fmt.Sprintf("%s:%s", tagName, v))
+		}
+		instanceID := fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
+		instanceIPs[lookupIP] = &gostatsd.Instance{
+			ID:   instanceID,
+			Tags: tags,
+		}
+		p.logger.WithFields(logrus.Fields{
+			"instance": instanceID,
+			"ip":       lookupIP,
+			"tags":     tags,
+		}).Debug("Added tags")
+	}
+	return instanceIPs, returnErr
+}
+
+// MaxInstancesBatch returns maximum number of instances that could be requested via the Instance method.
+func (p *Provider) MaxInstancesBatch() int {
+	// This is arbitrary since we have a local cache of information
+	return 64
+}
+
+// Name returns the name of the provider.
+func (p *Provider) Name() string {
+	return ProviderName
+}
+
+// SelfIP returns host's IPv4 address.
+func (p *Provider) SelfIP() (gostatsd.IP, error) {
+	// Get our pod's IP address from the network stack
+	podSelfIP, err := getSelfIPAddress()
+	if err != nil {
+		return gostatsd.UnknownIP, err
+	}
+	return podSelfIP, nil
+}
+
+// NewProviderFromOptions returns a new k8s provider.
+func NewProviderFromOptions(options gostatsd.CloudProviderOptions) (gostatsd.CloudProvider, error) {
+	a := util.GetSubViper(options.Viper, "k8s")
+	a.SetDefault(ParamKubeconfigContext, DefaultKubeconfigContext)
+	a.SetDefault(ParamKubeconfigPath, DefaultKubeconfigPath)
+
+	// Set up the k8s client
+	var restConfig *rest.Config
+	var err error
+	kubeconfigPath := a.GetString(ParamKubeconfigPath)
+	kubeconfigContext := a.GetString(ParamKubeconfigContext)
+	if kubeconfigPath != "" {
+		configOverrides := &clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: ""}}
```

Comment:
`ClusterInfo` does not need to be set? It's a noop, right?

Reply:
After testing it seems you're correct. Will remove.